### PR TITLE
Add Grmhd riemann problems analytic data

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Add the GRMHD Riemann problems as an analytic data. The exact solution is not easy to compute.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.cpp [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  p | left_rest_mass_density_;
+  p | right_rest_mass_density_;
+  p | left_pressure_;
+  p | right_pressure_;
+  p | left_spatial_velocity_;
+  p | right_spatial_velocity_;
+  p | left_magnetic_field_;
+  p | right_magnetic_field_;
+  p | equation_of_state_;
+  p | background_spacetime_;
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.cpp]: [nitpick] move eos and background to top of list so the pup order matches declaration order + initialization order, makes it easier to read.

File: src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.hpp [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+ * The standard problems were first collected in \cite Balsara2001. A complete
+ * Riemann solver for the RMHD equations is presented in \cite Giacomazzo2006
+ * and can be downloaded from https://www.brunogiacomazzo.org/?page_id=395
+ *
+ * The domain is from \f$[-0.5,0.5]^3\f$ with periodic boundary conditions in
+ * \f$y\f$ and \f$z\f$. The problems are usually run at a resolution of 1600
+ * finite difference/finite volume grid points. Below are the initial conditions
+ * for the 5 different Balsara Riemann problems. Please note that RP5 has a
+ * different final time than the rest, and that RP1 has a different adiabatic
+ * index than the rest.
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.hpp]: Would be helpful to add a discussion of the background spacetime. Why are the lapse and shift options if they have the same values in all these problems? why are they options if the background spacetime is set to minkowski always?

I would organize this as 3 short paragraph: First, the current sentences about the domain. Then, new sentences about the background. Last, the current sentences explaining the initial conditions from the reference.

File: src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.hpp [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+namespace grmhd::AnalyticData {
+/*!
+ * \brief Initial conditions for relativistic MHD Riemann problems
+ *
+ * The standard problems were first collected in \cite Balsara2001. A complete
+ * Riemann solver for the RMHD equations is presented in \cite Giacomazzo2006
+ * and can be downloaded from https://www.brunogiacomazzo.org/?page_id=395
+ *
+ * The domain is from \f$[-0.5,0.5]^3\f$ with periodic boundary conditions in
+ * \f$y\f$ and \f$z\f$. The problems are usually run at a resolution of 1600
```
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.hpp:36]: Say somewhere in the discussion of the domain that the discontinuity is initially at x = 0.

**Review by fmahebert — COMMENTED** [review]
Some nitpicks, feel free to squash directly.

**Comment by nilsdeppe** [comment]
I've squashed in the changes. Thanks for the review and let me know if the sentences I've added aren't clear :)

**Review by fmahebert — DISMISSED** [review]
LGTM, thanks!

File: docs/References.bib [hunk 4]
```diff
@@ -64,6 +64,21 @@ @article{Balsara1999
   adsnote =      {Provided by the SAO/NASA Astrophysics Data System}
 }
 
+@article{Balsara2001,
+  doi =          {10.1086/318941},
+  url =          {https://doi.org/10.1086/318941},
+  year =         2001,
+  month =        {jan},
```
Comment (kidder) [inline docs/References.bib]: `Jan` or remove the braces?

**Review by kidder — COMMENTED** [review]
you can squash

**Review by kidder — CHANGES_REQUESTED** [review]
not sure why my comment about the reference didn't make a request change

**Comment by nilsdeppe** [comment]
I've rebased and squashed in the braces removal. Thanks for reviewing :)

**Review by fmahebert — APPROVED** [review]

File: docs/References.bib (hunk 4 above)
Comment (kidder) [reply docs/References.bib]: This fixed the month, but now the year isn't rendering correctly, but I notice this happens for several entires and it is not obvious to me what is different about them.  So I suggest merging this PR and opening an issue about the incorrect rendering...

**Review by kidder — APPROVED** [review]

**Pull request merged**